File: Julia/8kyu/8kyu-area-or-perimeter.jl
# Area or Perimeter
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

# Solution 1
module Solution
  export areaOrPerimeter
  function areaOrPerimeter(l, w)
    ifelse(l == w, l * w, (l + w) * 2)
  end
end

# Solution 2 (Julia syntax)
module Solution
  export areaOrPerimeter
  function areaOrPerimeter(l, w)
    l == w ? l * w : (l + w) * 2
  end
end


# Tests
using FactCheck
facts("Basic tests") do
  @fact Solution.areaOrPerimeter(4, 4) --> 16
  @fact Solution.areaOrPerimeter(6, 10) --> 32
end

File: Python/8kyu/8kyu-area-or-perimeter.py
# Area or Perimeter
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

# Solution 1
def area_or_perimeter(l , w):
    return (l + w) * 2 if l != w else l * w

# Solution 2
def area_or_perimeter(l , w):
    return l * w if l == w else (l + w) * 2


# Tests
import codewars_test as test
from solution import area_or_perimeter

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(area_or_perimeter(4, 4), 16)
        test.assert_equals(area_or_perimeter(6, 10), 32)